guest.js

```diff
@@ -108,10 +107,8 @@ class ManyLevelGuest extends AbstractLevel {
       self[kFlushed]()
     })
 
-    const proxy = duplexify()
-    proxy.setWritable(decode)
-    proxy.setReadable(encode)
-    eos(proxy, cleanup)
+    const proxy = Duplex.from({ writable: decode, readable: encode })
```

Comment:
This requires node 16.8.0 and probably can't be browserified either because `readable-stream` is behind

Reply:
Although, on the browser point, perhaps it's time to let go of the old and build something web stream-based

Reply:
Sad but true... to both.

Reply:
Can we use `Duplex.from´ if available and otherwise fallback to `duplexify`?

Reply:
A few things I want to do first:

1. Benchmark `Duplex.from`, I didn't try that yet. And design a more suitable benchmark, because I've focused mainly on iterators so far, less on one-of operations like `put()` and `get()`. And swap `level` for `memory-level` to remove fs from the equation. And bypass `net` sockets.
2. Check compat with `length-prefixed-stream` which uses `readable-stream` and therefore doesn't satisfy core's `willEmitClose` logic. And `length-prefixed-stream` has its own `destroy()` 😞. Not sure what the effect is though, if any. Can you help with that?

Reply:
Indirectly that unfortunately includes any lib that uses them.